### ISSUE TITLE
antiderivative.PI_DeepONet_antideriv.PI_DeepONet.loss_operator

### ISSUE THREAD

**Issue by smao-astro** [PredictiveIntelligenceLab/Physics-informed-DeepONets#1]
Hi Sifan,

Thank you very much to make the excellent work public, I am trying to understand the code in `PI_DeepONet_antideriv.ipynb` and wondering would you like to provide some help on the `PI_DeepONet.loss_operator`.

Specifically, I have questions below:

- What is the loss term for? Constraint on initial condition or other thing?
- What is the admitted value of `y` and why? From the paper I think it should be just zeros, but when running the code I saw an 10^5 array with variable values, what is the purpose of these coordinate samples?

Thank you very much!

**Comment by sifanw094** [comment]
Hi,

1. What is the loss term for? Constraint on initial condition or other thing?
Yes, you are right. The loss PI_DeepONet.loss_operator aims to fit any data measurements, which include, but is not limited to, BC/IC conditions, numerical simulations, experimental data, etc.

2.  What is the admitted value of y and why? From the paper I think it should be just zeros, but when running the code I saw an 10^5 array with variable values, what is the purpose of these coordinate samples?
These coordinates are inputs of the trunk net at which the DeepONet output is evaluated.  In the paper, we performed two experiments about the antiderivative operator.  The first experiment aims to learn the antiderivative operator. So the training data is provided. The second experiment aims to solve the linear ODE system without any solution measurements.  In this case, y should be all zeros. If you carefully examine our code, then you would find the comment "**If solve the forward problem**". Below, we overwrite s and y with zeros.
